db-scripts/src/main/resources/migration.sql

```diff
@@ -806,7 +806,7 @@ AS (
     WHERE MUTATION_TYPE = 'fusion'
 );
 INSERT INTO genetic_profile(STABLE_ID, CANCER_STUDY_ID, GENETIC_ALTERATION_TYPE, DATATYPE, NAME, DESCRIPTION, SHOW_PROFILE_IN_ANALYSIS_TAB)
-SELECT CONCAT(CANCER_STUDY_IDENTIFIER, '_structural_variants'), CANCER_STUDY_ID, 'STRUCTURAL_VARIANT','SV','Fusions','Fusions',0
+SELECT CONCAT(CANCER_STUDY_IDENTIFIER, '_fusion'), CANCER_STUDY_ID, 'STRUCTURAL_VARIANT','SV','Fusions','Fusions',0
```

Comment:
Thanks for updating this! Just one issue: we have already applied this migration to one production database. 

I guess I can rerun this thing?

From: https://github.com/cBioPortal/cbioportal/pull/6945#issuecomment-568592112

```
delete from genetic_profile where STABLE_ID like '%_structural_variants%'
delete from sample_list_list where list_id in (select list_id from sample_list where CATEGORY = 'all_cases_with_sv_data')
delete from sample_list where CATEGORY = 'all_cases_with_sv_data'
```

Another thing - now that you've updated the sample service to support studies with a case list. Should we just remove this whole part of filling in case lists? I think it would be more comprehensible if running the migration on old studies with FUSION datafiles results in the same state of the database as importing the data. Right now from what I understand the migration creates a case list, but importing a study with FUSION data only adds a new profile.

Reply:
UPDATE: added note. Now that I think more about it. We could maybe get rid of the required migration and changes to the importer altogether if we change the service layer to return a `_fusion` profile if there are fusions in the mutation table. I don't think it would affect performance too much to find those. The advantage of that approach is that we're only touching the service layer, so:

- no migration necessary
- easier to remove the old FUSION related code since it's limited to only service layer

Reply:
you do not have to remove case list @inodb, just update the genetic_profile.STABLE_ID using following statement 
```
UPDATE genetic_profile
    INNER JOIN cancer_study ON cancer_study.CANCER_STUDY_ID = genetic_profile.CANCER_STUDY_ID
SET genetic_profile.STABLE_ID = CONCAT(CANCER_STUDY_IDENTIFIER, '_fusion')
WHERE genetic_profile.STABLE_ID like '%_structural_variants%' AND genetic_profile.datatype = 'FUSION' AND genetic_profile.genetic_alteration_type = 'STRUCTURAL_VARIANT'
```

Reply:
> Another thing - now that you've updated the sample service to support studies with a case list. Should we just remove this whole part of filling in case lists?

good catch, @inodb! we do not need backfilling case lists for fusions any more, but it will not hurt to leave the way as it is now as well.

Reply:
@khzhu agreed that keeping the case list migration wouldn't affect anything for the user, but for sake of clarity i think it would be cleaner to remove the backfilling. Because the state of the database will differ when you import the study from when you perform the migration. If e.g. somebody works on this code in the future they might assume the case lists to be there based on the migration code

What are your thoughts on removing the migration code & importer code altogether and changing the service layer instead to return a fusion profile if there's fusion data in the mutations table? Avoiding a database migration and importer update might make the rollout process easier

Reply:
There is ofc some performance cost to this, but it might be minor

Reply:
hi @inodb , sorry for a late comment. 

> Because the state of the database will differ when you import the study from when you perform the migration. If e.g. somebody works on this code in the future they might assume the case lists to be there based on the migration code

agree with you

> What are your thoughts on removing the migration code & importer code altogether and changing the service layer instead to return a fusion profile if there's fusion data in the mutations table? Avoiding a database migration and importer update might make the rollout process easier

probably not a good idea I would say. Genetic profile is an ORM object mapped to the genetic_profile entity in the database. Each genetic profile is corresponding to a row in the genetic profile table.

Reply:
@khzhu no worries! Thanks so much for still replying to this 😃 ! 

Let's change the migration script to not generate case lists then and leave creation of the fusion profile in there

Reply:
my pleasure, @inodb . Will make changes. thanks again for the review.

Reply:
Hi @inodb , migration script updated. please let me know if there is anything else need to be addressed. Thank you!